cmake/target_flags.cmake

```diff
@@ -30,6 +30,13 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*)")
   set(AARCH64 ON)
 endif()
 
+# in case of cross-compilation (or -m32) CMAKE_SYSTEM_PROCESSOR is equal to
+# CMAKE_HOST_SYSTEM_PROCESSOR which is X86_64; patch this until a better solution
+if(CMAKE_SIZEOF_VOID_P EQUAL 4 AND X86_64)
+  unset(X86_64)
+  set(X86 ON)
+endif()
```

Comment:
Why not to create a toolchain file? This condition is also true in case of cross compilation on `x86_64` but for `armhf`, or not?

Reply:
yes, this is valid for arm as well. 

How toolchain will change the picture?)

Reply:
let's keep current code as WA. Will be changed in future